### PR TITLE
Implement the feed feature 

### PR DESCRIPTION
**Pull request by MariemBaccari**
This PR implements the feed fragment which is very similar to the explore fragment except it only displays the reviews of the users you follow. It also refactors the database a little bit to add a LocationDatabase class that has the location methods to lighten the UserDatabase one because of codeclimate comments.

Screenshot_20230505-125641_MediaTo

### CODE REVIEW

**Comment by MariemBaccari** [comment]
Screenshot_20230505-125641_MediaTo

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import com.github.sdp.mediato.errorCheck.Preconditions;
+import com.github.sdp.mediato.model.Location;
+import com.github.sdp.mediato.model.User;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class LocationDatabase {
```
Comment (PH-14) [inline app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:14]: maybe add a javadoc to the class declaration

File: app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java [hunk 2]
```diff
@@ -367,5 +286,39 @@ public static CompletableFuture<List<User>> getAllUser(String username) {
         });
         return future;
     }
+
+    /**
+     * Gets all the users the user is following
+     * @param username of the reference user
+     * @return a completable future with a list of users
+     */
+    public static CompletableFuture<List<User>> getFollowingUsers(String username) {
```
Comment (PH-14) [inline app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java:295]: check if username isn't empty string

**Review by PH-14 — CHANGES_REQUESTED** [review]
Very good code and modularity, tests and comment are well written. Some changes requested due to preconditions tests not being made in the database classes, but it's not a lot

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java (hunk 1 above)
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:17]: Addressed in latest commit (even added it for the other database classes ;)

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    public static FirebaseDatabase database = FirebaseDatabase.getInstance();
+
+    /**
+     * Updates the user's location in the database
+     *
+     * @param username
+     * @param latitude
+     * @param longitude
+     */
+    public static void updateLocation(String username, double latitude, double longitude) {
```
[the remark this answers is not in the document]
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:28]: The location is already checked with `Preconditions.checkLocation(location)` for the right values. I added the check for the username.

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    /**
+     * Retrieves the user's saved location from the database
+     *
+     * @param username
+     * @return a completable future with the location in it
+     * @see Location class to check for validity
+     */
+    public static CompletableFuture<Location> getSavedLocation(String username) {
```
[the remark this answers is not in the document]
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:43]: done in latest commit

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    /**
+     * Gets all the nearby users' usernames
+     * @param username of the reference user
+     * @param radius in which we want to look for users
+     * @return a completable future with a list of strings containing the usernames
+     * @Note For now, DEFAULT_RADIUS is used instead of the radius parameter because the settings aren't implemented yet
+     */
+    public static CompletableFuture<List<String>> getNearbyUsernames(String username, double radius) {
```
[the remark this answers is not in the document]
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:65]: done in latest commit

File: app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        return future;
+    }
+
+    /**
+     * Gets all the nearby users
+     * @param username of the reference user
+     * @param radius in which we want to look for users
+     * @return a completable future with a list of users
+     */
+    public static CompletableFuture<List<User>> getNearbyUsers(String username, double radius) {
```
[the remark this answers is not in the document]
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/LocationDatabase.java:90]: done in latest commit

File: app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java (hunk 2 above)
Comment (MariemBaccari) [reply app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java:298]: done in latest commit

**Review by PH-14 — APPROVED** [review]
LGTM now !

**Pull request merged**